irc.py

```diff
@@ -257,20 +261,35 @@ def command(self, cmd: bytes) -> None:
             print('Unknown command: ', cmd)
 
 
-
 def main():
-    sl_client = slack.Slack()
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-p', '--port', type=int, action='store', dest='port',
+                                default=9007, required=False,
+                                help='set port number')
+    parser.add_argument('-i', '--ip', type=str, action='store', dest='ip',
+                                default='127.0.0.1', required=False,
+                                help='set ip address')
+    parser.add_argument('-t', '--tokenfile', type=str, action='store', dest='tokenfile',
+                                default=expanduser('~')+'/.localslackcattoken',
+                                required=False,
+                                help='set the token file')
+    parser.add_argument('-u', '--nouserlist', action='store_true',
+                                dest='nouserlist', required=False,
+                                help='don\'t display userlist')
+    args = parser.parse_args()
+
+    sl_client = slack.Slack(args)
     sl_events = sl_client.events_iter()
     serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     serversocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-    serversocket.bind(('127.0.0.1', 9007))
+    serversocket.bind((args.ip, args.port))
```

Comment:
do you think it'd be a good idea to have a warning or error if the ip doesn't begin with 127? (not local). This thing supports no authentication or encryption whatsoever.

Reply:
made the recommended changes. added -o to override the exit on non 127 address, that might be useful to someone at some point maybe.